pyshelf/endpoint_decorators.py

```diff
@@ -153,4 +154,25 @@ def wrapper(container, *args, **kwargs):
 
         return wrapper
 
+    def decode_request(self, func):
+        """
+            Will decode and inject the data if it is valid json
+            otherwise a error will be returned.
```

Comment:
This is really splitting hairs but technically a string, int, or boolean on its own valid json, right? ex. json.loads('"json-string"'). I know we have no use for that but should we make note of that in doc block?